apps/election-manager/src/utils/semsTallies.ts

```diff
@@ -0,0 +1,334 @@
+import _ from 'lodash'
+
+import {
+  Candidate,
+  CandidateContest,
+  Contest,
+  Dictionary,
+  Election,
+  YesNoContest,
+} from '@votingworks/types'
+
+import {
+  ContestOptionTally,
+  ContestTally,
+  ExternalTally,
+  FullElectionExternalTally,
+  TallyCategory,
+  YesNoOption,
+} from '../config/types'
+import assert from './assert'
+import { expandEitherNeitherContests, getContests } from './election'
+
+const WriteInCandidateId = '0'
+const OvervoteCandidateId = '1'
+const UndervoteCandidateId = '2'
+
+const writeInCandidate: Candidate = {
+  id: '__write-in',
+  name: 'Write-In',
+  isWriteIn: true,
+}
+
+export interface SEMSFileRow {
+  countyId: string
+  precinctId: string
+  contestId: string
+  contestTitle: string
+  partyId: string
+  partyName: string
+  candidateId: string
+  candidateName: string
+  candidatePartyId: string
+  candidatePartyName: string
+  numberOfVotes: number
+}
+
+// TODO(caro) revisit how to count the total number of ballots for multi seat contests
+// The number of total ballots is undervotes + overvotes + (othervotes / numseats)
+// That formula assumes all votes voted for the maxiumum number of seats allowed which is
+// probably not true in practice. This is irreleveant with the number of seats is 1.
+export function getContestTallyForCandidateContest(
+  contest: CandidateContest,
+  rows: SEMSFileRow[]
+): ContestTally {
+  const tallies: Dictionary<ContestOptionTally> = {}
+  let undervotes = 0
+  let overvotes = 0
+  let numVotes = 0
+  let writeInVotes = 0
+  const validCandidates = _.groupBy(contest.candidates, 'id')
+  for (const row of rows) {
+    if (row.candidateId === UndervoteCandidateId) {
+      undervotes = row.numberOfVotes
+      numVotes += row.numberOfVotes
+    } else if (row.candidateId === OvervoteCandidateId) {
+      overvotes = row.numberOfVotes
+      numVotes += row.numberOfVotes
+    } else if (
+      contest.allowWriteIns &&
+      row.candidateId === WriteInCandidateId
+    ) {
+      writeInVotes += row.numberOfVotes
+      numVotes += row.numberOfVotes / contest.seats
```

Comment:
this division feels like it should be done only once, maybe at the end of the counting?